codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/awsjson/AwsJson1_0_ProtocolGenerator.kt

```diff
@@ -32,9 +33,9 @@ open class AwsJson1_0_ProtocolGenerator : AWSHttpBindingProtocolGenerator() {
     override val httpResponseGenerator = HttpResponseGenerator(
         unknownServiceErrorSymbol,
         defaultTimestampFormat,
-        AWSJsonHttpResponseBindingErrorGenerator(),
+        HttpResponseBindingOutputGenerator(),
```

Comment:
Q: Before `HttpResponseBindingOutputGenerator()` was added to constructor call for `HttpResponseGenerator`, how were serializing / deserializing responses handled in codegen?

Reply:
Before, this type was hard-coded for every protocol.  Now there is a separate XML version of it since XML code generation is so dramatically different.

So, depending on whether the AWS protocol uses XML in responses, either the `XMLHttpResponseBindingOutputGenerator` or original `HttpResponseBindingOutputGenerator` is injected when the `HttpResponseGenerator` is created.